wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift

```diff
@@ -110,10 +110,16 @@ public struct SearchRequest {
     let searchDomain: String?
     let searchOptions: SearchOptions
 
-    public init(query: String, searchOptions: SearchOptions, team: Team? = nil) {
+    public init(
+        query: String,
+        searchOptions: SearchOptions,
+        team: Team? = nil,
+        isOtherDomainAllowed: Bool,
+        selfDomain: String
+    ) {
         let (query, searchDomain) = Self.parseQuery(query)
         self.query = query
-        self.searchDomain = searchDomain
+        self.searchDomain = isOtherDomainAllowed ? searchDomain : selfDomain
```

Comment:
Does this have to be part of the search request? Or can we do this in the code that performs the request?